symbolic-base/src/ZkFold/Symbolic/Data/Eq.hs

```diff
@@ -1,39 +1,38 @@
-{-# LANGUAGE DerivingStrategies   #-}
-{-# LANGUAGE TypeOperators        #-}
-{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE TypeOperators      #-}
 
 {-# OPTIONS_GHC -Wno-orphans #-}
 
 module ZkFold.Symbolic.Data.Eq (
     Eq(..),
     elem,
-    SymbolicEq,
     GEq (..)
 ) where
 
+import           Control.Monad                    (return)
 import           Data.Bool                        (bool)
+import           Data.Function                    (($))
 import           Data.Functor.Rep                 (mzipRep, mzipWithRep)
 import           Data.Traversable                 (for)
-import           Prelude                          (return, type (~), ($))
 import qualified Prelude                          as Haskell
 
 import           ZkFold.Algebra.Class
 import           ZkFold.Data.Eq
 import           ZkFold.Data.Package
 import           ZkFold.Symbolic.Class
-import           ZkFold.Symbolic.Data.Bool        (Bool (Bool), all, any)
+import           ZkFold.Symbolic.Data.Bool        (Bool (..), all, any)
 import           ZkFold.Symbolic.Data.Class
 import           ZkFold.Symbolic.Data.Combinators (runInvert)
 import           ZkFold.Symbolic.Data.Conditional ()
 import           ZkFold.Symbolic.MonadCircuit
 
 -- TODO: move to ZkFold.Symbolic.Data.Bool
 
-instance (Symbolic c, LayoutFunctor f) => Eq (c f) where
-    type BooleanOf (c f) = Bool c
+instance (SymbolicData x, Symbolic c) => Eq (x c) where
```

Comment:
A user might want to have a custom Eq instance, so I think an `instance Eq (Sym f c)` might be a better fit